### PR TITLE
WFLY-4835 remove references to ClassIndex

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java [hunk 1]
```diff
@@ -94,7 +94,7 @@
      */
     private final Set<Method> lifecycleMethods = new HashSet<>();
 
-    public ComponentConfiguration(final ComponentDescription componentDescription, final ClassIndex classIndex, final ClassLoader moduleClassLoader, final ModuleLoader moduleLoader) {
+    public ComponentConfiguration(final ComponentDescription componentDescription, final ClassReflectionIndex classIndex, final ClassLoader moduleClassLoader, final ModuleLoader moduleLoader) {
```
Comment (jamezp) [inline ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java]: Should this be `ClassReflectionIndex<?>` for consistency?

File: ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java [hunk 2]
```diff
@@ -505,7 +505,7 @@ public void setInstanceFactory(final ComponentFactory instanceFactory) {
         this.instanceFactory = instanceFactory;
     }
 
-    public ClassIndex getClassIndex() {
+    public ClassReflectionIndex getClassIndex() {
```
Comment (jamezp) [inline ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java]: Should this be `ClassReflectionIndex<?>` for consistency?

Comment (stuartwdouglas) [inline ee/src/main/java/org/jboss/as/ee/component/ComponentConfiguration.java]: Really ClassReflectionIndex should not be parameterised. I will remove all the <?> from full, so I can make the change in core.

**Comment by jamezp** [comment]
Checkstyle :(

**Comment by stuartwdouglas** [comment]
retest this please

**Comment by stuartwdouglas** [comment]
Retest this please

**Pull request merged**